From 8c857c1e72a145cd9c0904379615d09d4b9d1d98 Mon Sep 17 00:00:00 2001
From: Vincent Povirk <vincent@codeweavers.com>
Date: Wed, 13 Jun 2012 16:10:15 -0500
Subject: [PATCH 55/55] Ship MonoPosixHelper.dll.

---
 build-winemono.sh |    4 ++++
 mono              |    2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/build-winemono.sh b/build-winemono.sh
index 5fca71c..f80c3ca 100755
--- a/build-winemono.sh
+++ b/build-winemono.sh
@@ -63,12 +63,16 @@ cross_build_mono ()
         sed -e 's/-lgcc_s//' -i libtool
     fi
     WINEPREFIX=/dev/null make $MAKEOPTS || exit 1
+    cd "$CURDIR/build-cross-$ARCH/support"
+    WINEPREFIX=/dev/null make $MAKEOPTS || exit 1
     rm -rf "$CURDIR/build-cross-$ARCH-install"
+    cd "$CURDIR/build-cross-$ARCH"
     make install || exit 1
     cd "$CURDIR"
 
     mkdir -p "$CURDIR/image/bin"
     cp "$CURDIR/build-cross-$ARCH-install/bin/libmono-2.0.dll" "$CURDIR/image/bin/libmono-2.0-$ARCH.dll"
+    cp "$CURDIR/build-cross-$ARCH/support/.libs/libMonoPosixHelper.dll" "$CURDIR/image/bin/MonoPosixHelper-$ARCH.dll"
 }
 
 build_cli ()
