--- wine-mono-4.5.2/mono/configure.in.orig	2014-06-24 21:12:59.841794023 -0500
+++ wine-mono-4.5.2/mono/configure.in	2014-06-24 23:26:13.675935649 -0500
@@ -117,6 +117,8 @@
 			AC_DEFINE(MINGW_CROSS_COMPILE,1,[Cross-compiling using MinGW])
 		fi
 		HOST_CC="gcc"
+		CC="$host-gcc -static-libgcc"
+		CXX="$host-g++ -static-libgcc"
 		# Windows 2000 is required that includes Internet Explorer 5.01
 		CPPFLAGS="$CPPFLAGS -DWINVER=0x0500 -D_WIN32_WINNT=0x0500 -D_WIN32_IE=0x0501 -D_UNICODE -DUNICODE -DWIN32_THREADS -DFD_SETSIZE=1024 -DGC_NOT_DLL"
 		LDFLAGS="$LDFLAGS -lmswsock -lws2_32 -lole32 -loleaut32 -lpsapi -lversion -ladvapi32 -lwinmm -lkernel32"
--- wine-mono-4.5.2/mono/eglib/configure.ac.orig	2014-06-24 21:55:52.707897667 -0500
+++ wine-mono-4.5.2/mono/eglib/configure.ac	2014-06-24 23:28:01.488241190 -0500
@@ -57,6 +57,8 @@
     SEARCHSEP=';'
     OS="WIN32"
     PIDTYPE='void *'
+    CC="$host-gcc -static-libgcc"
+    CXX="$host-g++ -static-libgcc"
     ;;
 *-*darwin*)
     platform_darwin=yes
