--- build-winemono.sh.orig	2012-06-26 21:09:02.185640667 +0200
+++ build-winemono.sh	2012-06-26 21:14:24.272370920 +0200
@@ -157,25 +157,25 @@
 
 build_directorytable ()
 {
-    echo 'Directory\tDirectory_Parent\tDefaultDir'
-    echo 's72\tS72\tl255'
-    echo 'Directory\tDirectory'
-
-    echo 'TARGETDIR\t\tSourceDir'
-    echo 'MONODIR\tMONOBASEDIR\tmono-2.0:.'
-    echo 'MONOBASEDIR\tWindowsFolder\tmono:.'
-    echo 'WindowsFolder\tTARGETDIR\t.'
-    echo 'WindowsDotNet\tWindowsFolder\tMicrosoft.NET'
-    echo 'WindowsDotNetFramework\tWindowsDotNet\tFramework'
-    echo 'WindowsDotNetFramework11\tWindowsDotNetFramework\tv1.1.4322'
-    echo 'WindowsDotNetFramework11Config\tWindowsDotNetFramework11\tCONFIG'
-    echo 'WindowsDotNetFramework20\tWindowsDotNetFramework\tv2.0.50727'
-    echo 'WindowsDotNetFramework20Config\tWindowsDotNetFramework20\tCONFIG'
-    echo 'WindowsDotNetFramework30\tWindowsDotNetFramework\tv3.0'
-    echo 'WindowsDotNetFramework30wcf\tWindowsDotNetFramework30\twindows communication foundation'
-    echo 'WindowsDotNetFramework30wpf\tWindowsDotNetFramework30\twpf'
-    echo 'WindowsDotNetFramework40\tWindowsDotNetFramework\tv4.0.30319'
-    echo 'WindowsDotNetFramework40Config\tWindowsDotNetFramework40\tCONFIG'
+    echo -e 'Directory\tDirectory_Parent\tDefaultDir'
+    echo -e 's72\tS72\tl255'
+    echo -e 'Directory\tDirectory'
+
+    echo -e 'TARGETDIR\t\tSourceDir'
+    echo -e 'MONODIR\tMONOBASEDIR\tmono-2.0:.'
+    echo -e 'MONOBASEDIR\tWindowsFolder\tmono:.'
+    echo -e 'WindowsFolder\tTARGETDIR\t.'
+    echo -e 'WindowsDotNet\tWindowsFolder\tMicrosoft.NET'
+    echo -e 'WindowsDotNetFramework\tWindowsDotNet\tFramework'
+    echo -e 'WindowsDotNetFramework11\tWindowsDotNetFramework\tv1.1.4322'
+    echo -e 'WindowsDotNetFramework11Config\tWindowsDotNetFramework11\tCONFIG'
+    echo -e 'WindowsDotNetFramework20\tWindowsDotNetFramework\tv2.0.50727'
+    echo -e 'WindowsDotNetFramework20Config\tWindowsDotNetFramework20\tCONFIG'
+    echo -e 'WindowsDotNetFramework30\tWindowsDotNetFramework\tv3.0'
+    echo -e 'WindowsDotNetFramework30wcf\tWindowsDotNetFramework30\twindows communication foundation'
+    echo -e 'WindowsDotNetFramework30wpf\tWindowsDotNetFramework30\twpf'
+    echo -e 'WindowsDotNetFramework40\tWindowsDotNetFramework\tv4.0.30319'
+    echo -e 'WindowsDotNetFramework40Config\tWindowsDotNetFramework40\tCONFIG'
 
     cd "$CURDIR/image"
 
@@ -189,7 +189,7 @@
             DIRNAME=MONODIR
         fi
         BASENAME=`basename $f`
-        echo $KEY\\t$DIRNAME\\t$BASENAME
+        echo -e $KEY\\t$DIRNAME\\t$BASENAME
     done
 
     cd "$CURDIR"
@@ -197,24 +197,24 @@
 
 build_componenttable ()
 {
-    echo 'Component\tComponentId\tDirectory_\tAttributes\tCondition\tKeyPath'
-    echo 's72\tS38\ts72\ti2\tS255\tS72'
-    echo 'Component\tComponent'
-
-    echo 'mono-registry\t{93BE4304-497C-4ACB-A0FD-1C3695C011B4}\tWindowsDotNetFramework\t4\t\tDotNetFrameworkInstallRoot'
-    echo 'config-1.1\t{0DA29B5A-2050-4200-92EE-442D1EE6CF96}\tWindowsDotNetFramework11Config\t0\t\t1.1-machine.config'
-    echo 'config-2.0\t{ABB0BF6A-6610-4E45-8194-64D596667621}\tWindowsDotNetFramework20Config\t0\t\t2.0-machine.config'
-    echo 'config-4.0\t{511C0294-4504-4FC9-B5A7-E85CCEE95C6B}\tWindowsDotNetFramework40Config\t0\t\t4.0-machine.config'
-    echo 'dotnet-folder\t{22DCE198-F30F-4E74-AEC6-D089B844A878}\tWindowsDotNet\t0\t\t' # needed to remove the folder
-    echo 'framework-folder\t{41B3A67B-63F4-4491-A53C-9E792BE5A889}\tWindowsDotNetFramework\t0\t\t'
-    echo 'framework11-folder\t{20F5741D-4655-400D-8373-7607A84D2478}\tWindowsDotNetFramework11\t0\t\t'
-    echo 'framework20-folder\t{B845FD54-09B7-467C-800F-205A142F2F20}\tWindowsDotNetFramework20\t0\t\t'
-    echo 'framework30-folder\t{C3221C80-F9D2-41B5-91E1-F6ADBB05ABBC}\tWindowsDotNetFramework30\t0\t\t'
-    echo 'framework30wcf-folder\t{1ECAD22C-31C2-4BAC-AC74-78883C396FAB}\tWindowsDotNetFramework30wcf\t0\t\t'
-    echo 'framework30wpf-folder\t{3C146462-0CAF-4F07-83E6-A75A2A5DE961}\tWindowsDotNetFramework30wpf\t0\t\t'
-    echo 'framework40-folder\t{29ECF991-3E9E-4D23-B0B2-874631642B13}\tWindowsDotNetFramework40\t0\t\t'
-    echo 'monobase-folder\t{BE46D94A-7443-4B5C-9B91-6A83815365AB}\tMONOBASEDIR\t0\t\t'
-    echo 'mono-folder\t{FD7F9172-4E35-4DF5-BD6A-FB7B795D9346}\tMONODIR\t0\t\t'
+    echo -e 'Component\tComponentId\tDirectory_\tAttributes\tCondition\tKeyPath'
+    echo -e 's72\tS38\ts72\ti2\tS255\tS72'
+    echo -e 'Component\tComponent'
+
+    echo -e 'mono-registry\t{93BE4304-497C-4ACB-A0FD-1C3695C011B4}\tWindowsDotNetFramework\t4\t\tDotNetFrameworkInstallRoot'
+    echo -e 'config-1.1\t{0DA29B5A-2050-4200-92EE-442D1EE6CF96}\tWindowsDotNetFramework11Config\t0\t\t1.1-machine.config'
+    echo -e 'config-2.0\t{ABB0BF6A-6610-4E45-8194-64D596667621}\tWindowsDotNetFramework20Config\t0\t\t2.0-machine.config'
+    echo -e 'config-4.0\t{511C0294-4504-4FC9-B5A7-E85CCEE95C6B}\tWindowsDotNetFramework40Config\t0\t\t4.0-machine.config'
+    echo -e 'dotnet-folder\t{22DCE198-F30F-4E74-AEC6-D089B844A878}\tWindowsDotNet\t0\t\t' # needed to remove the folder
+    echo -e 'framework-folder\t{41B3A67B-63F4-4491-A53C-9E792BE5A889}\tWindowsDotNetFramework\t0\t\t'
+    echo -e 'framework11-folder\t{20F5741D-4655-400D-8373-7607A84D2478}\tWindowsDotNetFramework11\t0\t\t'
+    echo -e 'framework20-folder\t{B845FD54-09B7-467C-800F-205A142F2F20}\tWindowsDotNetFramework20\t0\t\t'
+    echo -e 'framework30-folder\t{C3221C80-F9D2-41B5-91E1-F6ADBB05ABBC}\tWindowsDotNetFramework30\t0\t\t'
+    echo -e 'framework30wcf-folder\t{1ECAD22C-31C2-4BAC-AC74-78883C396FAB}\tWindowsDotNetFramework30wcf\t0\t\t'
+    echo -e 'framework30wpf-folder\t{3C146462-0CAF-4F07-83E6-A75A2A5DE961}\tWindowsDotNetFramework30wpf\t0\t\t'
+    echo -e 'framework40-folder\t{29ECF991-3E9E-4D23-B0B2-874631642B13}\tWindowsDotNetFramework40\t0\t\t'
+    echo -e 'monobase-folder\t{BE46D94A-7443-4B5C-9B91-6A83815365AB}\tMONOBASEDIR\t0\t\t'
+    echo -e 'mono-folder\t{FD7F9172-4E35-4DF5-BD6A-FB7B795D9346}\tMONODIR\t0\t\t'
 
     cd "$CURDIR/image"
 
@@ -228,7 +228,7 @@
         fi
         GUID=`cat "$CURDIR/component-guids/${KEY}.guid"`
         KEYPATH=`find "$f" -maxdepth 1 -type f|sort|head -n 1|sed -e 's/\//!/g'`
-        echo $KEY\\t{$GUID}\\t$KEY\\t0\\t\\t$KEYPATH
+        echo -e $KEY\\t{$GUID}\\t$KEY\\t0\\t\\t$KEYPATH
     done
 
     cd "$CURDIR"
@@ -236,20 +236,20 @@
 
 build_createfoldertable ()
 {
-    echo 'Directory_\tComponent_'
-    echo 's72\ts72'
-    echo 'CreateFolder\tDirectory_\tComponent_'
-
-    echo 'WindowsDotNet\tdotnet-folder'
-    echo 'WindowsDotNetFramework\tframework-folder'
-    echo 'WindowsDotNetFramework11\tframework11-folder'
-    echo 'WindowsDotNetFramework20\tframework20-folder'
-    echo 'WindowsDotNetFramework30\tframework30-folder'
-    echo 'WindowsDotNetFramework30wcf\tframework30wcf-folder'
-    echo 'WindowsDotNetFramework30wpf\tframework30wpf-folder'
-    echo 'WindowsDotNetFramework40\tframework40-folder'
-    echo 'MONOBASEDIR\tmonobase-folder'
-    echo 'MONODIR\tmono-folder'
+    echo -e 'Directory_\tComponent_'
+    echo -e 's72\ts72'
+    echo -e 'CreateFolder\tDirectory_\tComponent_'
+
+    echo -e 'WindowsDotNet\tdotnet-folder'
+    echo -e 'WindowsDotNetFramework\tframework-folder'
+    echo -e 'WindowsDotNetFramework11\tframework11-folder'
+    echo -e 'WindowsDotNetFramework20\tframework20-folder'
+    echo -e 'WindowsDotNetFramework30\tframework30-folder'
+    echo -e 'WindowsDotNetFramework30wcf\tframework30wcf-folder'
+    echo -e 'WindowsDotNetFramework30wpf\tframework30wpf-folder'
+    echo -e 'WindowsDotNetFramework40\tframework40-folder'
+    echo -e 'MONOBASEDIR\tmonobase-folder'
+    echo -e 'MONODIR\tmono-folder'
 
     cd "$CURDIR/image"
 
@@ -260,7 +260,7 @@
         FILE=`find "$f" -maxdepth 1 -type f`
         if test ! "$FILE"; then
             KEY=`echo $f|sed -e 's/\//|/g'`
-            echo $KEY\\t$KEY
+            echo -e $KEY\\t$KEY
         fi
     done
 
@@ -269,24 +269,24 @@
 
 build_featurecomponentstable ()
 {
-    echo 'Feature_\tComponent_'
-    echo 's38\ts72'
-    echo 'FeatureComponents\tFeature_\tComponent_'
-
-    echo 'wine_mono\tmono-registry'
-    echo 'wine_mono\tconfig-1.1'
-    echo 'wine_mono\tconfig-2.0'
-    echo 'wine_mono\tconfig-4.0'
-    echo 'wine_mono\tdotnet-folder'
-    echo 'wine_mono\tframework-folder'
-    echo 'wine_mono\tframework11-folder'
-    echo 'wine_mono\tframework20-folder'
-    echo 'wine_mono\tframework30-folder'
-    echo 'wine_mono\tframework30wcf-folder'
-    echo 'wine_mono\tframework30wpf-folder'
-    echo 'wine_mono\tframework40-folder'
-    echo 'wine_mono\tmonobase-folder'
-    echo 'wine_mono\tmono-folder'
+    echo -e 'Feature_\tComponent_'
+    echo -e 's38\ts72'
+    echo -e 'FeatureComponents\tFeature_\tComponent_'
+
+    echo -e 'wine_mono\tmono-registry'
+    echo -e 'wine_mono\tconfig-1.1'
+    echo -e 'wine_mono\tconfig-2.0'
+    echo -e 'wine_mono\tconfig-4.0'
+    echo -e 'wine_mono\tdotnet-folder'
+    echo -e 'wine_mono\tframework-folder'
+    echo -e 'wine_mono\tframework11-folder'
+    echo -e 'wine_mono\tframework20-folder'
+    echo -e 'wine_mono\tframework30-folder'
+    echo -e 'wine_mono\tframework30wcf-folder'
+    echo -e 'wine_mono\tframework30wpf-folder'
+    echo -e 'wine_mono\tframework40-folder'
+    echo -e 'wine_mono\tmonobase-folder'
+    echo -e 'wine_mono\tmono-folder'
 
     cd "$CURDIR/image"
 
@@ -295,7 +295,7 @@
             continue
         fi
         KEY=`echo $f|sed -e 's/\//|/g'`
-        echo wine_mono\\t$KEY
+        echo -e wine_mono\\t$KEY
     done
 
     cd "$CURDIR"
@@ -303,9 +303,9 @@
 
 build_filetable ()
 {
-    echo 'File\tComponent_\tFileName\tFileSize\tVersion\tLanguage\tAttributes\tSequence'
-    echo 's72\ts72\tl255\ti4\tS72\tS20\tI2\ti2'
-    echo 'File\tFile'
+    echo -e 'File\tComponent_\tFileName\tFileSize\tVersion\tLanguage\tAttributes\tSequence'
+    echo -e 's72\ts72\tl255\ti4\tS72\tS20\tI2\ti2'
+    echo -e 'File\tFile'
 
     SEQ=0
 
@@ -334,7 +334,7 @@
         ;;
         esac
 
-        echo $KEY\\t$COMPONENT\\t$BASENAME\\t$FILESIZE\\t\\t\\t\\t$SEQ
+        echo -e $KEY\\t$COMPONENT\\t$BASENAME\\t$FILESIZE\\t\\t\\t\\t$SEQ
     done
 
     IMAGECAB_SEQ=$SEQ
@@ -344,11 +344,11 @@
 
 build_mediatable ()
 {
-    echo 'DiskId\tLastSequence\tDiskPrompt\tCabinet\tVolumeLabel\tSource'
-    echo 'i2\ti4\tL64\tS255\tS32\tS72'
-    echo 'Media\tDiskId'
+    echo -e 'DiskId\tLastSequence\tDiskPrompt\tCabinet\tVolumeLabel\tSource'
+    echo -e 'i2\ti4\tL64\tS255\tS32\tS72'
+    echo -e 'Media\tDiskId'
 
-    echo 1\\t$IMAGECAB_SEQ\\t\\t#image.cab\\t\\t
+    echo -e 1\\t$IMAGECAB_SEQ\\t\\t#image.cab\\t\\t
 }
 
 build_msi ()
