diff --git a/build-winemono.sh b/build-winemono.sh
index bee0dbf..7f0b062 100755
--- a/build-winemono.sh
+++ b/build-winemono.sh
@@ -120,6 +120,14 @@
         "$SRCDIR"/mono/configure --prefix="$BUILDDIR/build-cross-cli-install" --with-mcs-docs=no --disable-system-aot || exit 1
         sed -e 's+compiler_flags="-specs=/usr/lib/rpm/redhat/redhat-hardened-ld"+compiler_flags="-static"+' -i libtool
     fi
+
+	# put a System.Native library somewhere monolite can find it during the build
+	cd "$BUILDDIR/build-cross-cli/mono/native"
+	make $MAKEOPTS || exit 1
+	mkdir "$BUILDDIR/build-cross-cli/mono/lib/"
+	cp .libs/libmono-native.so "$BUILDDIR/build-cross-cli/mono/lib/libSystem.Native.so" || exit 1
+
+    cd "$BUILDDIR/build-cross-cli"
     if test 1 = $USE_MONOLITE; then
         make get-monolite-latest || exit 1
     elif test -e $SRCDIR/monolite/mcs.exe; then
